internal/services/pam/pam.go

```diff
@@ -138,6 +138,12 @@ func (s Service) SelectBroker(ctx context.Context, req *authd.SBRequest) (resp *
 		lang = "C"
 	}
 
+	// Throw an error if the user trying to authenticate already exists in the database and is disabled
+	userIsDisabled, err := s.userManager.IsUserDisabled(username)
+	if err == nil && userIsDisabled {
+		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("user %s is disabled", username))
+	}
+
```

Comment:
Let's handle unexpected errors here

```suggestion
	userIsDisabled, err := s.userManager.IsUserDisabled(username)
	if err != nil && !errors.Is(err, users.NoDataFoundError{}) {
		return nil, fmt.Errorf("could not check if user %q is disabled: %w", username, err)
	}
	if err == nil && userIsDisabled {
		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("user %s is disabled", username))
	}
```

Reply:
Thanks for suggesting. I have incorporated this change.